packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -738,6 +731,101 @@ class TextSelectionOverlay {
   }
 }
 
+/// This widget represents a text selection toolbar.
+class _SelectionToolbarOverlay extends StatefulWidget {
```

Comment:
Nit: Is there a better name for this class?  Or maybe just describe what it does in the comment above in a bit more detail.  I just wouldn't understand the difference between _SelectionToolbarOverlay and TextSelectionOverlay at a glance.  And it doesn't seem to have to do with Overlay directly.

I really like the idea of splitting this out into its own class like this by the way, it seems cleaner to me.

Reply:
_SelectionToolbarPlacement?  Something like that, naming is hard...

Reply:
For this I just followed the naming scheme of `_SelectionHandleOverlay` since this will be the widget created when we call

` _toolbar = OverlayEntry(builder: _buildToolbar);` , and `_buildToolbar` will return `_SelectionToolbarOverlay`.

to build the toolbar Overlay.

Reply:
Ah right, that makes sense 👍